www/docs/src/content/docs/cli/dep-ids.md

```diff
@@ -0,0 +1,200 @@
+# Dependency Identifiers (DepIds)
+
+Dependency Identifiers (DepIds) are a special string format used
+throughout the VLT package manager to uniquely identify dependencies.
+They are used as keys in the lockfile and other internal data
+structures.
+
+## Format Overview
+
+DepIds use a special delimiter character (`·`, Unicode MIDDLE DOT) to
+separate different parts of the identifier. This character was chosen
+because:
+
+- It's visually distinct
+- It's unlikely to appear in package names or versions
+- It's easy to distinguish in diffs and logs
+- It's a single Unicode character, making it efficient to process
+
+## DepId Types
+
+### 1. Registry Dependencies
+
+Registry dependencies are packages installed from a package registry
+(like npm).
+
+#### Default Registry
+
+```json
+"··name@version"
+```
+
+Example: `"··express@5.1.0"`
+
+#### Named Registry
+
+```json
+"·registry-name·name@version"
+```
+
+Example: `"·npm·lodash@4.17.21"`
+
+#### Custom Registry URL
+
+```json
+"·http%3A%2F%2Fvlt.sh%2F·x@1.2.3"
+```
+
+### 2. Git Dependencies
+
+Git dependencies are packages installed directly from a Git
+repository.
+
+```json
+"git·remote·selector"
+```
+
+Example: `"git·github:user/project·branchname"`
+
+The `selector` can be:
+
+- Branch name: `main`, `develop`
+- Tag: `v1.0.0`
+- Commit hash: `a1b2c3d`
+- Semver range: `^1.0.0`
+
+### 3. Workspace Dependencies
+
+Workspace dependencies are packages within the same monorepo.
+
+```json
+"workspace·path"
+```
+
+Example: `"workspace·src/mything"`
+
+### 4. File Dependencies
+
+File dependencies are packages installed from a local file or
+directory.
+
+```json
+"file·path"
+```
+
+Example: `"file·./local-package"`
+
+### 5. Remote Dependencies
+
+Remote dependencies are packages installed from a URL.
+
+```json
+"remote·url"
+```
+
+Example: `"remote·https%3A%2F%2Fexample.com"`
```

Comment:
That is not how we encode `/` characters, btw, because various systems get upset about folder paths containing `%2F`, as it looks like security shenanigans.

To get around this, we use the `§` section character as a stand-in for spaces.

```
> d.joinDepIDTuple(['remote', 'https://example.com/foo.tgz'])
'remote·https%3A§§example.com§foo.tgz'
```